From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matthias Maennich <maennich@google.com>
Date: Tue, 23 Feb 2021 12:49:45 +0000
Subject: ANDROID: add initial set of OWNERS

In order to provide better reviewer suggestions and to introduce fine
grained additional approval permissions, sprinkle some OWNERS files over
the tree.

This essentially grants additional OWNERS permissions in
  - net/
  - arch/arm*/net
  - arch/x86/net
  - drivers/net
  - fs/crypto
  - fs/f2fs
  - fs/verity

All other permissions are already covered by ref/meta/config:OWNERS.

Signed-off-by: Matthias Maennich <maennich@google.com>
Change-Id: I02a6563e4dd63842af37c50f7180bf38e9b9b4be
---
 Documentation/ABI/testing/OWNERS                 |  1 +
 OWNERS                                           | 12 ++++++++++++
 arch/arm/OWNERS                                  |  1 +
 arch/arm64/OWNERS                                |  7 +++++++
 arch/x86/OWNERS                                  |  3 +++
 crypto/OWNERS                                    |  1 +
 drivers/OWNERS                                   |  6 ++++++
 fs/OWNERS                                        |  3 +++
 fs/incfs/OWNERS                                  |  2 ++
 include/OWNERS                                   |  1 +
 include/linux/OWNERS                             |  1 +
 include/uapi/linux/OWNERS                        |  1 +
 kernel/sched/OWNERS                              |  4 ++++
 mm/OWNERS                                        |  3 +++
 net/OWNERS                                       |  2 ++
 tools/testing/selftests/filesystems/incfs/OWNERS |  1 +
 16 files changed, 49 insertions(+)
 create mode 100644 Documentation/ABI/testing/OWNERS
 create mode 100644 OWNERS
 create mode 100644 arch/arm/OWNERS
 create mode 100644 arch/arm64/OWNERS
 create mode 100644 arch/x86/OWNERS
 create mode 100644 crypto/OWNERS
 create mode 100644 drivers/OWNERS
 create mode 100644 fs/OWNERS
 create mode 100644 fs/incfs/OWNERS
 create mode 100644 include/OWNERS
 create mode 100644 include/linux/OWNERS
 create mode 100644 include/uapi/linux/OWNERS
 create mode 100644 kernel/sched/OWNERS
 create mode 100644 mm/OWNERS
 create mode 100644 net/OWNERS
 create mode 100644 tools/testing/selftests/filesystems/incfs/OWNERS

diff --git a/Documentation/ABI/testing/OWNERS b/Documentation/ABI/testing/OWNERS
new file mode 100644
--- /dev/null
+++ b/Documentation/ABI/testing/OWNERS
@@ -0,0 +1 @@
+per-file sysfs-fs-f2fs=file:/fs/f2fs/OWNERS
diff --git a/OWNERS b/OWNERS
new file mode 100644
--- /dev/null
+++ b/OWNERS
@@ -0,0 +1,12 @@
+# The full list of approvers is defined in
+# https://android.googlesource.com/kernel/common/+/refs/meta/config/OWNERS
+
+# The following OWNERS are defined at the top level to improve the OWNERS
+# suggestions through any user interface. Consider those people the ones that
+# can help with finding the best person to review.
+adelva@google.com
+gregkh@google.com
+maennich@google.com
+saravanak@google.com
+surenb@google.com
+tkjos@google.com
diff --git a/arch/arm/OWNERS b/arch/arm/OWNERS
new file mode 100644
--- /dev/null
+++ b/arch/arm/OWNERS
@@ -0,0 +1 @@
+include ../arm64/OWNERS
diff --git a/arch/arm64/OWNERS b/arch/arm64/OWNERS
new file mode 100644
--- /dev/null
+++ b/arch/arm64/OWNERS
@@ -0,0 +1,7 @@
+per-file crypto/*=file:/crypto/OWNERS
+per-file include/*=mzyngier@google.com,willdeacon@google.com
+per-file kernel/*=mzyngier@google.com,willdeacon@google.com
+per-file kvm/*=mzyngier@google.com,willdeacon@google.com
+per-file lib/*=mzyngier@google.com,willdeacon@google.com
+per-file mm/*=file:/mm/OWNERS
+per-file net/*=file:/net/OWNERS
diff --git a/arch/x86/OWNERS b/arch/x86/OWNERS
new file mode 100644
--- /dev/null
+++ b/arch/x86/OWNERS
@@ -0,0 +1,3 @@
+per-file crypto/*=file:/crypto/OWNERS
+per-file mm/*=file:/mm/OWNERS
+per-file net/*=file:/net/OWNERS
diff --git a/crypto/OWNERS b/crypto/OWNERS
new file mode 100644
--- /dev/null
+++ b/crypto/OWNERS
@@ -0,0 +1 @@
+ardb@google.com
diff --git a/drivers/OWNERS b/drivers/OWNERS
new file mode 100644
--- /dev/null
+++ b/drivers/OWNERS
@@ -0,0 +1,6 @@
+per-file base/*=gregkh@google.com,saravanak@google.com
+per-file block/*=akailash@google.com
+per-file md/*=akailash@google.com,palmerdabbelt@google.com,paullawrence@google.com
+per-file net/*=file:/net/OWNERS
+per-file tty/*=gregkh@google.com
+per-file usb/*=gregkh@google.com
diff --git a/fs/OWNERS b/fs/OWNERS
new file mode 100644
--- /dev/null
+++ b/fs/OWNERS
@@ -0,0 +1,3 @@
+per-file f2fs/*=jaegeuk@google.com
+per-file crypto/*=ebiggers@google.com
+per-file verity/*=ebiggers@google.com
diff --git a/fs/incfs/OWNERS b/fs/incfs/OWNERS
new file mode 100644
--- /dev/null
+++ b/fs/incfs/OWNERS
@@ -0,0 +1,2 @@
+akailash@google.com
+paullawrence@google.com
diff --git a/include/OWNERS b/include/OWNERS
new file mode 100644
--- /dev/null
+++ b/include/OWNERS
@@ -0,0 +1 @@
+per-file net=file:/net/OWNERS
diff --git a/include/linux/OWNERS b/include/linux/OWNERS
new file mode 100644
--- /dev/null
+++ b/include/linux/OWNERS
@@ -0,0 +1 @@
+per-file net*=file:/net/OWNERS
diff --git a/include/uapi/linux/OWNERS b/include/uapi/linux/OWNERS
new file mode 100644
--- /dev/null
+++ b/include/uapi/linux/OWNERS
@@ -0,0 +1 @@
+per-file net*=file:/net/OWNERS
diff --git a/kernel/sched/OWNERS b/kernel/sched/OWNERS
new file mode 100644
--- /dev/null
+++ b/kernel/sched/OWNERS
@@ -0,0 +1,4 @@
+connoro@google.com
+elavila@google.com
+qperret@google.com
+tkjos@google.com
diff --git a/mm/OWNERS b/mm/OWNERS
new file mode 100644
--- /dev/null
+++ b/mm/OWNERS
@@ -0,0 +1,3 @@
+hridya@google.com
+kaleshsingh@google.com
+surenb@google.com
diff --git a/net/OWNERS b/net/OWNERS
new file mode 100644
--- /dev/null
+++ b/net/OWNERS
@@ -0,0 +1,2 @@
+lorenzo@google.com
+maze@google.com
diff --git a/tools/testing/selftests/filesystems/incfs/OWNERS b/tools/testing/selftests/filesystems/incfs/OWNERS
new file mode 100644
--- /dev/null
+++ b/tools/testing/selftests/filesystems/incfs/OWNERS
@@ -0,0 +1 @@
+file:/fs/incfs/OWNERS
